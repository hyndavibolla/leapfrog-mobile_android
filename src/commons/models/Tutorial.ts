import { ReactNode } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export interface Tutorial {
  isParent?: boolean;
  isHidden?: boolean;
  parentId?: number;
  scrollToDown?: number;
}

export interface TutorialElementPosition extends Tutorial {
  x: number;
  y: number;
  children: ReactNode;
  style: StyleProp<ViewStyle>;
}
