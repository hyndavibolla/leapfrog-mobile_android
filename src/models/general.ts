import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
