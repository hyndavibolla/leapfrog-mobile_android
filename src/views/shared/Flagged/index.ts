export { default as Flagged, shouldShowFeature } from './Flagged';
