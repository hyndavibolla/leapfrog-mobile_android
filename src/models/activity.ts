import { IOffer } from './offer';

import { DateLike } from './general';

export enum Status {
  AVAILABLE = 'available',
  PENDING_CONFIRMATION = 'pendingConfirmation',
  POINT_BURN = 'pointBurn'
}

export enum Type {
  TRANSACTION = 'transaction',
  ADJUSTMENT = 'adjustment',
  REDEMPTION = 'redemption',
  REDEEM = 'redeem',
  EXPIRY = 'pointsexpiry',
  AVAILABLE = 'pointsavailable',
  RETURN = 'return',
  OTHER = 'other'
}

export interface IActivity {
  points: number;
  activityType: Type;
  requestorName: string;
  offers: IOffer[];
  timestamp: DateLike;
  grossSpend?: number;
  brandDetails?: {
    brandName: string;
    brandShortDescription: string;
    brandLogo: string;
  };
  txnId: string;
  activityDetails?: {
    partnerId: string;
    giftCardName: string;
    giftCardValue?: number;
  };
}

export const getFallbackActivity = (): IActivity => ({
  points: 0,
  activityType: Type.TRANSACTION,
  requestorName: '...',
  offers: [],
  timestamp: undefined,
  grossSpend: undefined,
  txnId: ''
});

export const getStatusDisplayText = (activityType: string, offer: IOffer): string => {
  if (!offer.pointStartDate || new Date(offer.pointStartDate).getTime() <= Date.now()) return 'Available';
  if (new Date(offer.pointStartDate).getTime() > Date.now()) return 'Pending';
  return activityType;
};
