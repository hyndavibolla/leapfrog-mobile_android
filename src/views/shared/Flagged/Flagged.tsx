import React, { memo, PropsWithChildren } from 'react';

import { FeatureFlag } from '../../../models/general';
import { ENV } from '../../../constants';
import ErrorBoundary from '../ErrorBoundary';

export const shouldShowFeature = (feature: FeatureFlag) => !ENV.IGNORED_FEATURE_LIST.includes(feature);

export interface Props {
  feature: FeatureFlag;
}

export const Flagged = ({ feature, children }: PropsWithChildren<Props>) => {
  return <ErrorBoundary>{shouldShowFeature(feature) ? children : null}</ErrorBoundary>;
};

export default memo(Flagged);
