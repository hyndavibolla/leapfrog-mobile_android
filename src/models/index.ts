import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ActivityModel from './activity';
import * as OfferModel from './offer';
import * as GameModel from './game';
import * as MissionModel from './mission';
import * as RewardModel from './reward';
import * as StreakModel from './streak';
import * as PromptModel from './prompt';
import * as CardLink from './cardLink';

export { GeneralModel, UserModel, ActivityModel, OfferModel, GameModel, MissionModel, RewardModel, StreakModel, PromptModel, CardLink };
