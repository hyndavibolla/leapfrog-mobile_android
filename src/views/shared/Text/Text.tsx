import React, { memo, ReactNode } from 'react';
import { TextProps } from 'react-native';

import { COLOR, FONT_FAMILY, FONT_SIZE, LINE_HEIGHT } from '_constants';

import { StyledText } from './styles';

export interface Props extends TextProps {
  color?: COLOR;
  font?: FONT_FAMILY;
  size?: FONT_SIZE;
  lineHeight?: LINE_HEIGHT;
  children: ReactNode;
}

const Text = ({ color, font, size, lineHeight, children, ...otherProps }: Props) => {
  return (
    <StyledText color={color} font={font} size={size} lineHeight={lineHeight} {...otherProps}>
      {children}
    </StyledText>
  );
};

export default memo(Text);
