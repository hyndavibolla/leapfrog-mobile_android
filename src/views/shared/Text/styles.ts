import styled from '@emotion/native';

import { FONT_FAMILY, COLOR, FONT_SIZE, LINE_HEIGHT } from '_constants';

import { Props } from './Text';

export const StyledText = styled.Text<Props>`
  font-family: ${({ font }: Props) => font || FONT_FAMILY.MEDIUM};
  color: ${({ color }: Props) => color || COLOR.DARK_GRAY};
  font-size: ${({ size }: Props) => size || FONT_SIZE.SMALLER};
  line-height: ${({ lineHeight }: Props) => lineHeight || LINE_HEIGHT.PARAGRAPH};
`;
