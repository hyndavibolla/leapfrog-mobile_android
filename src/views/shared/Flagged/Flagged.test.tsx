import React from 'react';
import { View } from 'react-native';
import { Text } from '../Text';
import { render } from '@testing-library/react-native';

import Flagged, { shouldShowFeature } from './Flagged';
import { FeatureFlag } from '../../../models/general';

describe('Flagged', () => {
  it('should tell if a feature should be shown', () => {
    expect(shouldShowFeature(FeatureFlag.TEST_ONLY)).toBeTruthy();
    expect(shouldShowFeature(FeatureFlag.ENV_BANNER)).toBeFalsy();
  });

  it('should show features depending on the env', () => {
    const { toJSON } = render(
      <View>
        <Flagged feature={FeatureFlag.TEST_ONLY}>
          <Text>this should be on the snap</Text>
        </Flagged>
        <Flagged feature={FeatureFlag.ENV_BANNER}>
          <Text>this should NOT be on the snap</Text>
        </Flagged>
      </View>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
