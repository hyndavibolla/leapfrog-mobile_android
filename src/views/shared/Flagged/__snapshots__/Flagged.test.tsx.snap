// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flagged should show features depending on the env 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "color": "#909BA0",
          "fontFamily": "SFProDisplay-Medium",
          "fontSize": 16,
          "lineHeight": 22,
        },
        undefined,
      ]
    }
  >
    this should be on the snap
  </Text>
</View>
`;
